packages/docregistry/src/context.ts

```diff
@@ -595,23 +595,26 @@ export class Context<
     this._saveState.emit('started');
     const model = this._model;
     let content: PartialJSONValue;
+    let source: string;
     if (this._factory.fileFormat === 'json') {
       content = model.toJSON();
+      source = JSON.stringify(content);
     } else {
       content = model.toString();
       if (this._lineEnding) {
         content = content.replace(/\n/g, this._lineEnding);
       }
+      source = content;
     }
 
     const options = {
       type: this._factory.contentType,
-      format: this._factory.fileFormat,
-      content
+      format: this._factory.fileFormat
     };
     try {
       let value: Contents.IModel;
       await this._manager.ready;
+      this._provider.putSource(source);
```

Comment:
We have to be aware that the websocket message might arrive after the HTTP PUT request (so you will safe the old content). But since this is just an intermediate step, this is fine (although not production-ready yet).